packages/ember-glimmer/tests/integration/binding_integration_test.js

```diff
@@ -0,0 +1,89 @@
+import { RenderingTest, moduleFor } from '../utils/test-case';
+import { Component } from '../utils/helpers';
+import { set } from 'ember-metal/property_set';
+import { Binding } from 'ember-metal/binding';
+
+moduleFor('Binding integration tests', class extends RenderingTest {
+
+  ['@test bound values should update after the parent element is removed and then re-appended']() {
+    let instance;
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        instance = this;
+      }
+    });
+
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template: '{{#if showStuff}}{{boundValue}}{{else}}Not true.{{/if}}' });
+    this.render('{{foo-bar showStuff=showStuff boundValue=boundValue}}', {
+      showStuff: true,
+      boundValue: 'foo'
+    });
+    this.assertText('foo');
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'showStuff', false));
+    this.assertText('Not true.');
+    this.runTask(() => set(this.context, 'showStuff', true));
+    this.assertText('foo');
+
+    this.runTask(() => {
+      instance.remove();
+      set(this.context, 'showStuff', false);
+    });
+    this.runTask(() => {
+      set(this.context, 'showStuff', true);
+      this.rerender();
+    });
+
+    this.runTask(() => set(this.context, 'boundValue', 'bar'));
+    this.assertText('bar');
+  }
+
+  ['@htmlbars should accept bindings as a string or an Ember.binding']() {
+    let FooBarComponent = Component.extend({
+      twoWayTestBinding: Binding.from('direction'),
+      stringTestBinding: 'direction',
+      twoWayObjectTestBinding: Binding.from('displacement.distance'),
+      stringObjectTestBinding: 'displacement.distance'
+    });
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: FooBarComponent,
+      template: 'two way: {{twoWayTest}}, string: {{stringTest}}, object: {{twoWayObjectTest}}, string object: {{stringObjectTest}}'
+    });
+
+    this.render('{{foo-bar direction=direction displacement=displacement}}', {
+      direction: 'down',
+      displacement: {
+        distance: 10
+      }
+    });
+
+    this.assertText('two way: down, string: down, object: 10, string object: 10');
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'direction',  'up'));
+
+    this.assertText('two way: up, string: up, object: 10, string object: 10');
+
+    this.runTask(() => set(this.context, 'displacement.distance', 20));
+
+    this.assertText('two way: up, string: up, object: 20, string object: 20');
+
+    this.runTask(() => {
+      set(this.context, 'direction', 'right');
+      set(this.context, 'displacement', { distance: 30 });
+    });
+
+    this.assertText('two way: right, string: right, object: 30, string object: 30');
+
+    this.runTask(() => {
+      set(this.context, 'direction', 'down');
+      set(this.context, 'displacement.distance', 10);
```

Comment:
Switch this to replacement (`set(this.context, 'displacement', { distance: 30 });`)